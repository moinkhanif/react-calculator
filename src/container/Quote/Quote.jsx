import QUOTES from './Quotes-data';
import './Quote.styles.css';

const Quote = () => {
  const [quote, author] = QUOTES[
    Math.floor(Math.random() * QUOTES.length)
  ].split('â€“');
  return (
    <blockquote id="quote">
      {quote}
      <span>{`- ${author}`}</span>
    </blockquote>
  );
};

export default Quote;
