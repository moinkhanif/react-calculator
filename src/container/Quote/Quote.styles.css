#quote {
  height: 50vh;
  width: 100%;
  max-width: 700px;
  margin: auto;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: 1.5rem;
  color: #9f5f80;
}

#quote span {
  margin-top: 30px;
  color: #ff8e71;
}
