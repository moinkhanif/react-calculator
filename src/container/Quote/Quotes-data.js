const QUOTES = [
  'Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. –William Paul Thurston',
  'Without mathematics, there’s nothing you can do. Everything around you is mathematics. Everything around you is numbers. –Shakuntala Devi',
  'Somehow it’s o.k. for people to chuckle about not being good at math. Yet if I said ‘I never learned to read,” they’d say I was an illiterate dolt.” –Neil Degrasse Tyson',
  'The only way to learn mathematics is to do mathematics. –Paul Halmos',
];

export default QUOTES;
