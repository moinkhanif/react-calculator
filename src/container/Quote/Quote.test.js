import renderer from 'react-test-renderer';
import Quote from './Quote';

test('renders learn react link', () => {
  const quoteData = ['Test Quote -Testie'];
  const result = renderer.create(<Quote quoteData={quoteData} />).toJSON();
  expect(result).toMatchSnapshot();
});
