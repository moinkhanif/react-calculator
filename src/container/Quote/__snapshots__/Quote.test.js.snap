// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<blockquote
  id="quote"
>
  Test Quote -Testie
  <span>
    - undefined
  </span>
</blockquote>
`;
